File: packages/node/src/cutout/cutout.ts
import { Requester } from "../http";
import { mapParams } from "../utils";
import {
  CutoutClothingBackgroundRemovalParams,
  CutoutClothingBackgroundRemovalResponse,
  CutoutClothingBackgroundRemovalResponseData,
  validateCutoutClothingBackgroundRemovalParams,
  CutoutClothingBackgroundRemovalParamMap,
  CutoutProductBackgroundRemovalParams,
  CutoutProductBackgroundRemovalResponse,
  CutoutProductBackgroundRemovalResponseData,
  validateCutoutProductBackgroundRemovalParams,
  CutoutProductBackgroundRemovalParamMap,
  CutoutFoodBackgroundRemovalParams,
  CutoutFoodBackgroundRemovalResponse,
  CutoutFoodBackgroundRemovalResponseData,
  validateCutoutFoodBackgroundRemovalParams,
  CutoutFoodBackgroundRemovalParamMap,
  CutoutUniversalBackgroundRemovalParams,
  CutoutUniversalBackgroundRemovalResponse,
  CutoutUniversalBackgroundRemovalResponseData,
  validateCutoutUniversalBackgroundRemovalParams,
  CutoutUniversalBackgroundRemovalParamMap,
  CutoutAvatarExtractionParams,
  CutoutAvatarExtractionResponse,
  CutoutAvatarExtractionResponseData,
  validateCutoutAvatarExtractionParams,
  CutoutAvatarExtractionParamMap,
  CutoutHairExtractionParams,
  CutoutHairExtractionResponse,
  CutoutHairExtractionResponseData,
  validateCutoutHairExtractionParams,
  CutoutHairExtractionParamMap,
  CutoutHumanBackgroundRemovalParams,
  CutoutHumanBackgroundRemovalResponse,
  CutoutHumanBackgroundRemovalResponseData,
  validateCutoutHumanBackgroundRemovalParams,
  CutoutHumanBackgroundRemovalParamMap
} from "../generated/endpoints";

export class CutoutAPI {
  constructor(private request: Requester) {}

  async cutoutClothingBackgroundRemoval(params: CutoutClothingBackgroundRemovalParams): Promise<CutoutClothingBackgroundRemovalResponse> {
    validateCutoutClothingBackgroundRemovalParams(params);
    const mapped = mapParams(params, CutoutClothingBackgroundRemovalParamMap);
    return this.request.request<CutoutClothingBackgroundRemovalResponseData>("POST", "/api/cutout/general/apparel-background-removal", { body: mapped, multipart: true });
  }

  async cutoutProductBackgroundRemoval(params: CutoutProductBackgroundRemovalParams): Promise<CutoutProductBackgroundRemovalResponse> {
    validateCutoutProductBackgroundRemovalParams(params);
    const mapped = mapParams(params, CutoutProductBackgroundRemovalParamMap);
    return this.request.request<CutoutProductBackgroundRemovalResponseData>("POST", "/api/cutout/general/commodity-background-removal", { body: mapped, multipart: true });
  }

  async cutoutFoodBackgroundRemoval(params: CutoutFoodBackgroundRemovalParams): Promise<CutoutFoodBackgroundRemovalResponse> {
    validateCutoutFoodBackgroundRemovalParams(params);
    const mapped = mapParams(params, CutoutFoodBackgroundRemovalParamMap);
    return this.request.request<CutoutFoodBackgroundRemovalResponseData>("POST", "/api/cutout/general/food-background-removal", { body: mapped, multipart: true });
  }

  async cutoutUniversalBackgroundRemoval(params: CutoutUniversalBackgroundRemovalParams): Promise<CutoutUniversalBackgroundRemovalResponse> {
    validateCutoutUniversalBackgroundRemovalParams(params);
    const mapped = mapParams(params, CutoutUniversalBackgroundRemovalParamMap);
    return this.request.request<CutoutUniversalBackgroundRemovalResponseData>("POST", "/api/cutout/general/universal-background-removal", { body: mapped, multipart: true });
  }

  async cutoutAvatarExtraction(params: CutoutAvatarExtractionParams): Promise<CutoutAvatarExtractionResponse> {
    validateCutoutAvatarExtractionParams(params);
    const mapped = mapParams(params, CutoutAvatarExtractionParamMap);
    return this.request.request<CutoutAvatarExtractionResponseData>("POST", "/api/cutout/portrait/avatar-extraction", { body: mapped, multipart: true });
  }

  async cutoutHairExtraction(params: CutoutHairExtractionParams): Promise<CutoutHairExtractionResponse> {
    validateCutoutHairExtractionParams(params);
    const mapped = mapParams(params, CutoutHairExtractionParamMap);
    return this.request.request<CutoutHairExtractionResponseData>("POST", "/api/cutout/portrait/hairstyle-extraction", { body: mapped, multipart: true });
  }

  async cutoutHumanBackgroundRemoval(params: CutoutHumanBackgroundRemovalParams): Promise<CutoutHumanBackgroundRemovalResponse> {
    validateCutoutHumanBackgroundRemovalParams(params);
    const mapped = mapParams(params, CutoutHumanBackgroundRemovalParamMap);
    return this.request.request<CutoutHumanBackgroundRemovalResponseData>("POST", "/api/cutout/portrait/portrait-background-removal", { body: mapped, multipart: true });
  }

}

File: packages/python/ailabtools/cutout.py
from .http import Requester
from .utils import map_params
from .generated.endpoints import (
    CutoutClothingBackgroundRemovalParams,
    CutoutClothingBackgroundRemovalResponse,
    CutoutClothingBackgroundRemovalParamMap,
    validate_cutoutClothingBackgroundRemoval_params,
    CutoutProductBackgroundRemovalParams,
    CutoutProductBackgroundRemovalResponse,
    CutoutProductBackgroundRemovalParamMap,
    validate_cutoutProductBackgroundRemoval_params,
    CutoutFoodBackgroundRemovalParams,
    CutoutFoodBackgroundRemovalResponse,
    CutoutFoodBackgroundRemovalParamMap,
    validate_cutoutFoodBackgroundRemoval_params,
    CutoutUniversalBackgroundRemovalParams,
    CutoutUniversalBackgroundRemovalResponse,
    CutoutUniversalBackgroundRemovalParamMap,
    validate_cutoutUniversalBackgroundRemoval_params,
    CutoutAvatarExtractionParams,
    CutoutAvatarExtractionResponse,
    CutoutAvatarExtractionParamMap,
    validate_cutoutAvatarExtraction_params,
    CutoutHairExtractionParams,
    CutoutHairExtractionResponse,
    CutoutHairExtractionParamMap,
    validate_cutoutHairExtraction_params,
    CutoutHumanBackgroundRemovalParams,
    CutoutHumanBackgroundRemovalResponse,
    CutoutHumanBackgroundRemovalParamMap,
    validate_cutoutHumanBackgroundRemoval_params,
    CutoutHDUniversalBackgroundRemovalParams,
    CutoutHDUniversalBackgroundRemovalResponse,
    CutoutHDUniversalBackgroundRemovalParamMap,
    validate_cutoutHDUniversalBackgroundRemoval_params,
    CutoutHdHumanBodyBackgroundRemovalParams,
    CutoutHdHumanBodyBackgroundRemovalResponse,
    CutoutHdHumanBodyBackgroundRemovalParamMap,
    validate_cutoutHdHumanBodyBackgroundRemoval_params
)

class CutoutAPI:
    def __init__(self, requester: Requester):
        self._requester = requester

    async def cutoutClothingBackgroundRemoval(self, params: CutoutClothingBackgroundRemovalParams) -> CutoutClothingBackgroundRemovalResponse:
        validate_cutoutClothingBackgroundRemoval_params(params)
        mapped = map_params(params, CutoutClothingBackgroundRemovalParamMap)
        return await self._requester.request('POST', '/api/cutout/general/apparel-background-removal', body=mapped, multipart=True)

    async def cutoutProductBackgroundRemoval(self, params: CutoutProductBackgroundRemovalParams) -> CutoutProductBackgroundRemovalResponse:
        validate_cutoutProductBackgroundRemoval_params(params)
        mapped = map_params(params, CutoutProductBackgroundRemovalParamMap)
        return await self._requester.request('POST', '/api/cutout/general/commodity-background-removal', body=mapped, multipart=True)

    async def cutoutFoodBackgroundRemoval(self, params: CutoutFoodBackgroundRemovalParams) -> CutoutFoodBackgroundRemovalResponse:
        validate_cutoutFoodBackgroundRemoval_params(params)
        mapped = map_params(params, CutoutFoodBackgroundRemovalParamMap)
        return await self._requester.request('POST', '/api/cutout/general/food-background-removal', body=mapped, multipart=True)

    async def cutoutUniversalBackgroundRemoval(self, params: CutoutUniversalBackgroundRemovalParams) -> CutoutUniversalBackgroundRemovalResponse:
        validate_cutoutUniversalBackgroundRemoval_params(params)
        mapped = map_params(params, CutoutUniversalBackgroundRemovalParamMap)
        return await self._requester.request('POST', '/api/cutout/general/universal-background-removal', body=mapped, multipart=True)

    async def cutoutAvatarExtraction(self, params: CutoutAvatarExtractionParams) -> CutoutAvatarExtractionResponse:
        validate_cutoutAvatarExtraction_params(params)
        mapped = map_params(params, CutoutAvatarExtractionParamMap)
        return await self._requester.request('POST', '/api/cutout/portrait/avatar-extraction', body=mapped, multipart=True)

    async def cutoutHairExtraction(self, params: CutoutHairExtractionParams) -> CutoutHairExtractionResponse:
        validate_cutoutHairExtraction_params(params)
        mapped = map_params(params, CutoutHairExtractionParamMap)
        return await self._requester.request('POST', '/api/cutout/portrait/hairstyle-extraction', body=mapped, multipart=True)

    async def cutoutHumanBackgroundRemoval(self, params: CutoutHumanBackgroundRemovalParams) -> CutoutHumanBackgroundRemovalResponse:
        validate_cutoutHumanBackgroundRemoval_params(params)
        mapped = map_params(params, CutoutHumanBackgroundRemovalParamMap)
        return await self._requester.request('POST', '/api/cutout/portrait/portrait-background-removal', body=mapped, multipart=True)

    async def cutoutHDUniversalBackgroundRemoval(self, params: CutoutHDUniversalBackgroundRemovalParams) -> CutoutHDUniversalBackgroundRemovalResponse:
        validate_cutoutHDUniversalBackgroundRemoval_params(params)
        mapped = map_params(params, CutoutHDUniversalBackgroundRemovalParamMap)
        return await self._requester.request('POST', '/api/cutout/general/hd-universal-background-removal', body=mapped, multipart=True)

    async def cutoutHdHumanBodyBackgroundRemoval(self, params: CutoutHdHumanBodyBackgroundRemovalParams) -> CutoutHdHumanBodyBackgroundRemovalResponse:
        validate_cutoutHdHumanBodyBackgroundRemoval_params(params)
        mapped = map_params(params, CutoutHdHumanBodyBackgroundRemovalParamMap)
        return await self._requester.request('POST', '/api/cutout/portrait/hd-portrait-background-removal', body=mapped, multipart=True)
